test/cli_test.py

```diff
@@ -930,7 +937,10 @@ def test_dict_show_get_clear(servicer, server_url_env, set_env_client):
     key = ("baz-dict", os.environ.get("MODAL_ENVIRONMENT", "main"))
     dict_id = "di-abc123"
     servicer.deployed_dicts[key] = dict_id
-    servicer.dicts[dict_id] = {dumps("a"): dumps(123), dumps("b"): dumps("blah")}
+    servicer.dicts[dict_id] = {
+        dumps("a", protocol=4): dumps(123, protocol=4),
```

Comment:
Why do we need `protocol=4` here suddenly?

Reply:
We hard code the protocol to 4 here: https://github.com/modal-labs/modal-client/blob/49a5f02c20766d722447d0e961bc2df098e53258/modal/_serialization.py#L32 and the default protocol in 3.14 changed to 5: https://docs.python.org/3/library/pickle.html#pickle.DEFAULT_PROTOCOL